### PR TITLE
Add ellipsified for table header and rows (see issue #4370)

### PR DESCRIPTION
**Pull request by joannekoong**
Addresses issue #4370 (for both table header and table rows -- see pics)

### CODE REVIEW

**Comment by joannekoong** [comment]
header_ellipsified

**Comment by joannekoong** [comment]
header_ellipsified

**Comment by joannekoong** [comment]
row_ellipsified

**Review by kdoh — CHANGES_REQUESTED** [review]
Hey there @joannekoong. Thanks for this! A couple quick bits of feedback.

- It looks like something is causing the sort indicator icon to wrap onto a new line (compared to master)
- The tooltips seem to be lacking padding in this instance.

File: frontend/src/metabase/visualizations/components/TableInteractive.jsx [hunk 1]
```diff
@@ -320,7 +321,7 @@ export default class TableInteractive extends Component<*, Props, State> {
                     className={cx("cellData", { "cursor-pointer": isSortable })}
                     onClick={isSortable && this.setSort.bind(this, column)}
                 >
-                    {columnTitle}
+                    <Ellipsified> {columnTitle} </Ellipsified>
```
Comment (kdoh) [inline frontend/src/metabase/visualizations/components/TableInteractive.jsx]: One other small thing would be removing the extra spaces in between the components here and on line 285.

**Review by kdoh — CHANGES_REQUESTED** [review]

**Comment by joannekoong** [comment]
Hi @kdoh, thanks for the comments. Great catch -- didn't realize the sort icon got kicked onto a new line. New pics are: 

header_ellipsified

row_ellipsified

File: frontend/src/metabase/visualizations/components/TableInteractive.jsx [hunk 2]
```diff
@@ -321,14 +321,16 @@ export default class TableInteractive extends Component<*, Props, State> {
...
                 >
-                    <Ellipsified> {columnTitle} </Ellipsified>
-                    {isSortable &&
-                        <Icon
-                            className="Icon ml1"
-                            name={sort && sort[0] && sort[0][1] === "ascending" ? "chevronup" : "chevrondown"}
-                            size={8}
-                        />
-                    }
+                    <Ellipsified><span>
```
Comment (kdoh) [inline frontend/src/metabase/visualizations/components/TableInteractive.jsx]: We'll want to keep the nesting consistent so It'd be great to wrap the spans onto their own line and then indent the children accordingly just like normal HTML.

**Review by kdoh — CHANGES_REQUESTED** [review]
A couple small stylistic details and then I think this is good to go.

File: frontend/src/metabase/css/components/popover.css [hunk 3]
```diff
@@ -26,6 +26,7 @@
 	font-weight: bold;
 	background-color: rgb(76,71,71);
 	border: none;
+	padding: 0.8em;
```
Comment (tlrobinson) [inline frontend/src/metabase/css/components/popover.css]: This is going to add extra padding to every existing tooltip. Please remove it and wrap `<Value>` in `<div className="p1">` (or maybe `p2`, see how it looks)

File: frontend/src/metabase/visualizations/components/TableInteractive.jsx [hunk 4]
```diff
@@ -281,7 +282,7 @@ export default class TableInteractive extends Component<*, Props, State> {
                     })}
                     onClick={isFilterable && this.showPopover.bind(this, rowIndex, columnIndex)}
                 >
-                    <Value value={cellData} column={column} onResize={this.onCellResize.bind(this, columnIndex)} />
+                    <Ellipsified><Value value={cellData} column={column} onResize={this.onCellResize.bind(this, columnIndex)} /></Ellipsified>
```
Comment (tlrobinson) [inline frontend/src/metabase/visualizations/components/TableInteractive.jsx]: This line is getting long, could you put `<Value>` on it's own line (and indent like the rest)

**Review by tlrobinson — CHANGES_REQUESTED** [review]

**Comment by joannekoong** [comment]
Pictures for "584688d: Incorporate tlrobinson's comments -- fix padding and add line breaks"

f
s

**Review by kdoh — APPROVED** [review]

**Comment by tlrobinson** [comment]
Unfortunately my advice was bad... adding the `<div>` with padding broke the ellipsification and vertical center alignment.

Ellipsified takes a `tooltip` prop. I think this is a workable solution:

```diff
diff --git a/frontend/src/metabase/visualizations/components/TableInteractive.jsx b/frontend/src/metabase/visualizations/components/TableInteractive.jsx
index 60ea357..e6c5d22 100644
--- a/frontend/src/metabase/visualizations/components/TableInteractive.jsx
+++ b/frontend/src/metabase/visualizations/components/TableInteractive.jsx
@@ -282,10 +282,8 @@ export default class TableInteractive extends Component<*, Props, State> {
                     })}
                     onClick={isFilterable && this.showPopover.bind(this, rowIndex, columnIndex)}
                 >
-                    <Ellipsified>
-                        <div className="p1">
-                            <Value value={cellData} column={column} onResize={this.onCellResize.bind(this, columnIndex)} />
-                        </div>
+                    <Ellipsified tooltip={<div className="p1"><Value value={cellData} column={column} /></div>}>
+                        <Value value={cellData} column={column} onResize={this.onCellResize.bind(this, columnIndex)} />
                     </Ellipsified>
                     { popover && popover.rowIndex === rowIndex && popover.columnIndex === columnIndex &&
                         <QuickFilterPopover
@@ -325,8 +323,8 @@ export default class TableInteractive extends Component<*, Props, State> {
                     className={cx("cellData", { "cursor-pointer": isSortable })}
                     onClick={isSortable && this.setSort.bind(this, column)}
                 >
-                    <Ellipsified>
-                        <div className="p1">
+                    <Ellipsified tooltip={columnTitle}>
+                        <span>
                             {columnTitle}
                             {isSortable &&
                                 <Icon
@@ -335,7 +333,7 @@ export default class TableInteractive extends Component<*, Props, State> {
                                     size={8}
                                 />
                             }
-                        </div>
+                        </span>
                     </Ellipsified>
                 </div>
                 <Draggable
```

**Comment by kdoh** [comment]
Hey there @joannekoong. Wanted to check in and see if you're still interested in finishing off this PR by making the changes @tlrobinson requested above.

**Comment by joannekoong** [comment]
Hi @kdoh, thanks for checking in -- I'm definitely interested in finishing this PR. Sorry for the delay, am swamped with school right now. I'll try my best to get this in by the end of the weekend!

**Comment by kdoh** [comment]
@tlrobinson mind giving this another look? looks like @joannekoong updated things.

**Comment by salsakran** [comment]
@tlrobinson + @kdoh how do you two want to proceed with this PR?

**Comment by tlrobinson** [comment]
I've opened a new PR with this merged with `master` here: https://github.com/metabase/metabase/pull/9067

**Comment by kdoh** [comment]
Implemented and merged in #9067

**Pull request closed without merging**